vowpalwabbit/core/src/reductions/lda_core.cc

```diff
@@ -1361,21 +1349,36 @@ base_learner* VW::reductions::lda_setup(VW::setup_base_i& stack_builder)
     all.example_parser = VW::make_unique<VW::parser>(minibatch2, previous_strict_parse);
   }
 
+  if (ld->minibatch > 1)
+  {
+    ld->saved_batch_examples.reserve(ld->minibatch);
+    for (uint64_t i = 0; i < ld->minibatch; i++)
+    {
+      ld->saved_batch_examples.emplace_back(VW::make_unique<VW::example>());
+    }
+  }
+
   ld->v.resize(all.lda * ld->minibatch);
 
   ld->decay_levels.push_back(0.f);
 
   all.example_parser->lbl_parser = VW::no_label_parser_global;
 
+  // If minibatch is > 1, then the predict function does not actually produce predictions.
```

Comment:
We should probably fix this before we merge? Or at the very least get a John opinion on if this is ok for the time being

Reply:
Its existed this way for a long time (probably since the beginning), we just correctly report the type now. That is, that it produces no prediction.

Reply:
Discussed offline, its fine since it is maintaining previous behavior